src/stores/account.ts

```diff
@@ -61,14 +98,117 @@ export default class AccountStore implements IAccountStore {
   };
 
   @action
-  changeAddress = (index: number) => {
-    if (index >= 0 && index < this.addresses.length) {
-      this.selectedAddress = this.addresses[index];
+  setActivationKey = (activationKey: string) => {
+    this.activationKey = activationKey;
+  };
+
+  @action
+  listKeyFiles = (): ProtectedPrivateKey[] => {
+    return listKeystoreFiles().map((keyId: string) => {
+      const key: V3Keystore = JSON.parse(
+        fs.readFileSync(path.resolve(sanitizeKeypath(), keyId), "utf8")
+      );
+      const ppk: ProtectedPrivateKey = {
+        keyId: key.id,
+        address: key.address,
+        path: path.resolve(sanitizeKeypath(), keyId),
+      };
+      return ppk;
+    });
+  };
+
+  @action
+  findKeyByAddress = (address: Address): Promise<ProtectedPrivateKey> => {
+    return new Promise<ProtectedPrivateKey>((resolve, reject) => {
+      const key = this.keyring.find((key) => {
+        return (
+          key.address.replace("0x", "") ===
+          address.replace("0x", "").toLowerCase()
+        );
+      });
+      if (key !== undefined) {
+        resolve(key);
+      } else {
+        reject(
+          Error(`No key file found matching with provided address: ${address}`)
+        );
+      }
+    });
+  };
+
+  //TODO: This function solely depending on behavior that
+  //addV3 push to end of the array, we need to fix that.
+  @action
+  importRaw = (privateKey: string, passphrase: string) => {
+    return rawPrivateKeyToV3(privateKey, passphrase).then((v3) => {
+      if (v3 !== undefined) {
+        const filePath = path.resolve(
+          sanitizeKeypath(),
+          [
+            "UTC--",
+            new Date().toJSON().replace(/:/g, "-").slice(0, -5),
+            "Z",
+            "--",
+            v3.id,
+          ].join("")
+        );
+        fs.writeFileSync(filePath, JSON.stringify(v3));
+        this.addKey({
+          keyId: v3.id,
+          address: v3.address,
+          path: filePath,
+        });
+        this.address = v3.address;
+        getAccountFromFile(v3.id, passphrase).then((account) =>
+          this.setAccount(account)
+        );
+      } else {
+        throw Error("Importing raw private key to V3 file failed.");
+      }
+    });
+  };
+
+  @action
+  isValidPrivateKey = (privateKey: string): boolean => {
+    try {
+      createAccount(privateKey);
+    } catch (e) {
+      return false;
     }
+    return true;
   };
```

Comment:
`isValidPrivateKey` function seems able to be separated as a non-class function.

Reply:
Seperated By adding `isValidPrivateKey` in @planetarium/account-localm